src/gui/transferlistwidget.cpp

```diff
@@ -589,9 +594,11 @@ void TransferListWidget::deleteSelectedTorrents(const bool deleteLocalFiles)
     if (Preferences::instance()->confirmTorrentDeletion()) {
         auto *dialog = new DeletionConfirmationDialog(this, torrents.size(), torrents[0]->name(), deleteLocalFiles);
         dialog->setAttribute(Qt::WA_DeleteOnClose);
-        connect(dialog, &DeletionConfirmationDialog::accepted, this, [dialog, torrents]()
+        connect(dialog, &DeletionConfirmationDialog::accepted, this, [this, dialog]()
         {
-            removeTorrents(torrents, dialog->isDeleteFileSelected());
+            // Some torrents might be removed when waiting for user input, so refetch the torrent list
```

Comment:
You don't need to call getSelectedTorrents above. Why to build list twice? Just check for `selectionModel()->hasSelection()`.
Something similar is in case of visible torrents.

Reply:
`DeletionConfirmationDialog` constructor requires some of the infos.

Reply:
Anyway `selectionModel()->selectedRows()->count()` is better, isn't it?

Reply:
I'm not following... are you talking about line 592? I was talking about line 595.

Reply:
Well, I see you need first torrent name. But seems not so optimal to extract all selected torrent handles for obtaining the name of the first one.

Reply:
>But seems not so optimal to extract all selected torrent handles for obtaining the name of the first one.

Yes, but the torrent list can also be used in line 606. We can really separate the code paths instead of using a common part, however I'm not sure that would be easier to maintain. And since this code isn't very frequently visited I have no strong opinion.

Reply:
Well, I won't push it any further.